posts/2023-08-22-ArgoCD-Configuration-Drift.adoc

```diff
@@ -0,0 +1,156 @@
+Now that you've successfully installed ArgoCD and gained access to its console, the next step is to set up a GitHub repository for Argo CD to synchronize with.
+
+For the purposes of this blog, we'll use the Daytrader7 sample application. This application is built around the concept of an online stock trading system, and the Daytrader7 README provides a comprehensive overview.
+
+It's a recommended best practice to maintain separation between configuration repositories and code repositories. Keeping these separate ensures that infrastructure changes are decoupled from application changes, allowing for more granular control, better audit trails, and reduced risk of unintended side-effects. For this example, we'll create a deploy directory within our existing repository to house the deployment configuration.
+
+To begin, navigate to GitHub and fork this repository: https://github.com/WASdev/sample.daytrader7. Use your forked repository when configuring ArgoCD in the following steps
+
+We assume that your CI pipleline has a way to build contianer image for your application. For this example, we have a pre-built cotnainer image used in the deployment files `deploy/daytrader7-deploy.yaml`.
+
+== Deploying DayTrader 7 via GitOps ==
+
+Now that we have the Git configuration repository ready for deployment, it’s time to configure ArgoCD to deploy the application.
+
+Go to the Argo CD console and ensure that you are logged in.
+Click the New App button near the top left of the console.
+For Application name, let’s call this one daytrader7, keeping it in the default Argo project and setting the sync policy to Manual. Check the Auto-Create Namespace box as well.
+[.img_border_light]
+image::/img/blog/configuration-drift-3.png[Creating the App,width=70%,align="center"]
+Scroll down to the Source and change the Repository URL to your newly forked repo. Change the path, which is the path where Argo CD is looking for deployment files, to deploy. Set the Cluster URL to https://kubernetes.default.svc, which is the local cluster URL. Then set the namespace to daytrader7
+Click Create in the top left.
+[.img_border_light]
+image::/img/blog/configuration-drift-4.png[Setting the Repo,width=70%,align="center"]
+
+== Syncing DayTrader ==
+On the main page of the Argo CD console, you should see a new tile that looks like this:
+[.img_border_light]
+image::/img/blog/configuration-drift-5.png[Argo Dashboard,width=70%,align="center"]
+Click the Sync button, then on the window that appears and click Sync again.
+Click on the daytrader7 tile to view the app dashboard and sync progress.
+Over the course of a few minutes, you should see resources being created and the app dashboard looking like this:
+[.img_border_light]
+image::/img/blog/configuration-drift-6.png[App deploying,width=70%,align="center"]
+Now the app is deployed and can be reached in a route created under the daytrader7 namespace. Find the URL via oc get routes –n daytrader7, then paste the URL in a browser.
```

Comment:
"oc get routes -n" should be formatted so we know it's a command and not just text/prose.